gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemBase.java

```diff
@@ -1508,14 +1508,22 @@ private Credential getCredential(Configuration config)
             .withPrefixes(CONFIG_KEY_PREFIXES)
             .get(config, config::get);
```

Comment:
You should move an above `IMPERSONATION_SERVICE_ACCOUNT_SUFFIX....get(...)` statement to be a last one that assigns to `serviceAccountToImpersonate` variable, because otherwise this generic impersonation configuration will override per-user/group configuration.

Reply:
I believe the current chain will do the job.

Reply:
Done.